File: clients/ios/Lextures/Core/LMS/LMSFeatureModelsPeopleAdmin.swift
import Foundation

// MARK: - People admin (M14.3)

struct PersonRow: Decodable, Identifiable, Hashable {
    var id: String
    var email: String
    var firstName: String?
    var lastName: String?
    var displayName: String?
    var orgId: String
    var orgName: String
    var role: String
    var active: Bool
    var createdAt: String
}

struct PaginatedPeople: Decodable {
    var items: [PersonRow]
    var total: Int64
    var page: Int
    var perPage: Int
    var totalPages: Int
}

struct PeopleDashboardStats: Decodable, Equatable {
    var signupsLast7Days: Int64
    var activeAccounts: Int64
    var totalAccounts: Int64
    var recentlyActive30Days: Int64
    var suspendedAccounts: Int64
}

enum PeopleListFilter: String, CaseIterable, Identifiable, Hashable {
    case signups7d = "signups_7d"
    case active
    case recent30d = "recent_30d"
    case total
    case suspended
    var id: String { rawValue }
}

struct PersonEnrollment: Decodable, Identifiable, Hashable {
    var courseId: String
    var courseCode: String
    var courseTitle: String
    var role: String
    var active: Bool
    var state: String
    var enrolledAt: String
    var orgName: String?

    var id: String { "\(courseId)-\(role)" }
}

struct PersonActivity: Decodable, Identifiable, Hashable {
    var eventKind: String
    var courseCode: String
    var courseTitle: String
    var occurredAt: String

    var id: String { "\(occurredAt)-\(eventKind)-\(courseCode)" }
}

struct PersonReport: Decodable {
    var id: String
    var email: String
    var firstName: String?
    var lastName: String?
    var displayName: String?
    var orgId: String
    var orgName: String
    var role: String
    var active: Bool
    var createdAt: String
    var lastActivityAt: String?
    var enrollmentCount: Int
    var enrollments: [PersonEnrollment]
    var recentActivity: [PersonActivity]
}

struct InvitePersonRequest: Encodable {
    var email: String
    var firstName: String?
    var lastName: String?
}

struct PatchPersonRequest: Encodable {
    var active: Bool
}

struct ForgotPasswordRequest: Encodable {
    var email: String
}

File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/LmsFeatureModelsPeopleAdmin.kt
package com.lextures.android.core.lms

import kotlinx.serialization.Serializable

@Serializable
data class PersonRow(
    val id: String,
    val email: String,
    val firstName: String? = null,
    val lastName: String? = null,
    val displayName: String? = null,
    val orgId: String,
    val orgName: String,
    val role: String,
    val active: Boolean,
    val createdAt: String,
)

@Serializable
data class PaginatedPeople(
    val items: List<PersonRow> = emptyList(),
    val total: Long = 0,
    val page: Int = 1,
    val perPage: Int = 25,
    val totalPages: Int = 0,
)

@Serializable
data class PeopleDashboardStats(
    val signupsLast7Days: Long = 0,
    val activeAccounts: Long = 0,
    val totalAccounts: Long = 0,
    val recentlyActive30Days: Long = 0,
    val suspendedAccounts: Long = 0,
)

enum class PeopleListFilter(val apiValue: String) {
    Signups7d("signups_7d"),
    Active("active"),
    Recent30d("recent_30d"),
    Total("total"),
    Suspended("suspended"),
}

@Serializable
data class PersonEnrollment(
    val courseId: String,
    val courseCode: String,
    val courseTitle: String,
    val role: String,
    val active: Boolean,
    val state: String,
    val enrolledAt: String,
    val orgName: String? = null,
)

@Serializable
data class PersonActivity(
    val eventKind: String,
    val courseCode: String,
    val courseTitle: String,
    val occurredAt: String,
)

@Serializable
data class PersonReport(
    val id: String,
    val email: String,
    val firstName: String? = null,
    val lastName: String? = null,
    val displayName: String? = null,
    val orgId: String,
    val orgName: String,
    val role: String,
    val active: Boolean,
    val createdAt: String,
    val lastActivityAt: String? = null,
    val enrollmentCount: Int = 0,
    val enrollments: List<PersonEnrollment> = emptyList(),
    val recentActivity: List<PersonActivity> = emptyList(),
)

@Serializable
data class InvitePersonRequest(
    val email: String,
    val firstName: String? = null,
    val lastName: String? = null,
)

@Serializable
data class PatchPersonRequest(
    val active: Boolean,
)

@Serializable
data class ForgotPasswordRequest(
    val email: String,
)
